src/main/java/com/code_intelligence/jazzer/runtime/ClojureLangHooks.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Code Intelligence GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.code_intelligence.jazzer.runtime;
+
+import com.code_intelligence.jazzer.api.HookType;
+import com.code_intelligence.jazzer.api.MethodHook;
+import java.lang.invoke.MethodHandle;
+import java.util.*;
+
+@SuppressWarnings("unused")
+final public class ClojureLangHooks {
+    static ThreadLocal<WeakHashMap<Object, String>> markedObjects = ThreadLocal.withInitial(() -> new WeakHashMap<Object, String>());
+
+    @MethodHook(type = HookType.AFTER, targetClassName = "clojure.lang.Var", targetMethod = "getRawRoot")
+    public static void clojureMarkContains(MethodHandle method, Object thisObject, Object[] arguments, int hookId, Object result) {
+        if ("clojure.string$includes_QMARK_".equals(result.getClass().getCanonicalName())
+                || "clojure.string$starts_with_QMARK_".equals(result.getClass().getCanonicalName())
+                || "clojure.string$ends_with_QMARK_".equals(result.getClass().getCanonicalName())
+                || "clojure.string$index_of_QMARK_".equals(result.getClass().getCanonicalName())
+                || "clojure.string$last_index_of_QMARK_".equals(result.getClass().getCanonicalName())) {
+            markedObjects.get().put(result, result.getClass().getCanonicalName());
+        }
+    }
+
+    @MethodHook(type = HookType.BEFORE, targetClassName = "clojure.lang.IFn", targetMethod = "invoke")
+    public static void clojureMarkedContains(
+            MethodHandle method, Object thisObject, Object[] args, int hookId) {
+        if (markedObjects.get().containsKey(thisObject)) {
```

Comment:
Are we using the value associated with the key in any way? If not, you can use `Collections.newSetFromMap` to craft a `WeakHashSet`.

Reply:
Not for now, but I could imagine that in the future we would like to hook different types of functions with different hook-behaviour. In this case, storing the name of the function in the HashMap would be helpful.